File: lib/src/main/java/com/orientsec/easysocket/utils/NoLogger.java
package com.orientsec.easysocket.utils;

/**
 * NoLogger is a no-op implementation of the Logger interface.
 * This class provides empty method implementations for all logging levels,
 * effectively disabling logging when used.
 */
class NoLogger implements Logger {
    @Override
    public void e(String msg) {
    }

    @Override
    public void e(String msg, Throwable t) {
    }

    @Override
    public void i(String msg) {
    }

    @Override
    public void i(String msg, Throwable t) {
    }

    @Override
    public void w(String msg) {
    }

    @Override
    public void w(String msg, Throwable t) {
    }

    @Override
    public void d(String msg) {
    }

    @Override
    public void d(String msg, Throwable t) {
    }
}

File: lib/src/main/kotlin/com/orientsec/easysocket/utils/NoLogger.kt
package com.orientsec.easysocket.utils

/**
 * NoLogger is a no-op implementation of the Logger interface.
 * This class provides empty method implementations for all logging levels,
 * effectively disabling logging when used.
 */
internal class NoLogger : Logger {
    override fun e(msg: String) {}

    override fun e(msg: String, t: Throwable?) {}

    override fun i(msg: String) {}

    override fun i(msg: String, t: Throwable?) {}

    override fun w(msg: String) {}

    override fun w(msg: String, t: Throwable?) {}

    override fun d(msg: String) {}

    override fun d(msg: String, t: Throwable?) {}
}
